core-primitives/stf-state-handler/src/in_memory_state_file_io.rs

```diff
@@ -204,13 +200,61 @@ where
 	}
 }
 
+#[cfg(feature = "sgx")]
+pub mod sgx {
+	use super::*;
+	use crate::{error::Result, file_io::sgx::list_shards};
+	use ita_sgx_runtime::Runtime;
+	use ita_stf::{Getter, Stf, TrustedCallSigned};
+	use itp_sgx_externalities::{SgxExternalities, SgxExternalitiesType};
+	use itp_stf_interface::InitState;
+	use itp_types::AccountId;
+	use std::sync::Arc;
+
+	type EnclaveStf = Stf<TrustedCallSigned, Getter, SgxExternalities, Runtime>;
+
+	pub fn create_in_memory_state_io_from_shards_directories(
+	) -> Result<Arc<InMemoryStateFileIo<SgxExternalitiesType, SgxExternalities>>> {
+		let shards = list_shards()?;
+		Ok(create_in_memory_externalities_state_io::<EnclaveStf>(&shards))
+	}
+
+	pub fn create_sgx_externalities_in_memory_state_io(
+	) -> Arc<InMemoryStateFileIo<SgxExternalitiesType, SgxExternalities>> {
+		create_in_memory_externalities_state_io::<EnclaveStf>(&[])
+	}
+
+	fn create_in_memory_externalities_state_io<Stf>(
+		shards: &[ShardIdentifier],
+	) -> Arc<InMemoryStateFileIo<SgxExternalitiesType, SgxExternalities>>
+	where
+		Stf: InitState<SgxExternalities, AccountId>,
+	{
+		Arc::new(InMemoryStateFileIo::new(
+			shards,
+			sgx_externalities_selector(),
+			Box::new(|| Stf::init_state(AccountId::new([1u8; 32])).state),
+			sgx_externalities_wrapper(),
+		))
+	}
+
+	fn sgx_externalities_selector() -> InnerStateSelector<SgxExternalitiesType, SgxExternalities> {
+		Box::new(|s| s.state)
+	}
+
+	fn sgx_externalities_wrapper() -> ExternalStateGenerator<SgxExternalitiesType, SgxExternalities>
+	{
+		Box::new(|s| SgxExternalities { state: s, state_diff: Default::default() })
```

Comment:
Is this ok to use the default state diff here instead of the one that comes with the state already?